### DISCUSSION TITLE
Incremental Site Generation - Remove pages

### DISCUSSION THREAD

**Discussion by marco910** [vercel/next.js#21709] [Help]
Hi everyone,

Since a few days I'm testing around with Incremental Site Generation in Next.js.

This is my current scenario:
- Next.js production build with Incremental Site Generation
- Prismic as headless CMS

In Prismic I'm creating from time to time new content which can be accessed by a dynamic route in Next.js. If create a new content, a page is generated from Next.js. So far, so good.
The problem I'm facing is when I delete content in Prismic. Then I want Next.js to delete the generated page also. Is there a way to do that?

Thanks in advance for your help!

**Comment by MehediH** [comment]
I have not used Prismic myself, but if your site is hosted on Vercel, you could do something like the following: 

* Set-up a webhook on Prismic when a document gets unpublished https://user-guides.prismic.io/en/articles/790505-webhooks
* This will then call the Vercel webhook which triggers a new deployment https://vercel.com/docs/more/deploy-hooks
